src/aspire/denoising/denoiser_cov2d.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, src, basis, var_noise, ctf_info):
             logger.info(f'CTF filters are not included in Cov2D denoising')
             # set all CTF filters to an identity filter
             self.ctf_idx = np.zeros(self.nimg, dtype=int)
-            self.ctf_fb = [blk_diag_eye(blk_diag_partition(RadialCTFFilter().fb_mat(basis)))]
+            self.ctf_fb = [BlkDiagMatrix.eye(get_partition(RadialCTFFilter().fb_mat(basis)))]
```

Comment:
Regarding the dtype, yes, thanks! I laughed, easy to miss yeah.

Otherwise, get_partition is exposed as a class staticmethod and a module function (just wraps the class staticmethod).

`BlkDiagMatrix.eye` needs just partition shapes.  I could make `zeros_like(B)` and `eye_like(B)` methods like numpy does.  I think that might be clearer, I'll push that in it's own commit and can decide then.

Reply:
(fwiw, in the latest develop rebase, I think this code no longer exists here in this file, instead implemented in `self.cov2d` which is BatchedRotCov2d, same idea though)..

Reply:
Replaced with `eye_like` method.